Marlin/SdFatUtil.cpp

```diff
@@ -33,15 +33,26 @@ int SdFatUtil::FreeRam() {
   return &top - reinterpret_cast<char*>(sbrk(0));
 }
 #else  // __arm__
-extern char *__brkval;
-extern char __bss_end;
+extern "C" {
+  extern int  __bss_end;
+  extern int* __brkval;
+};
 /** Amount of free RAM
  * \return The number of free bytes.
  */
 int SdFatUtil::FreeRam() {
-  char top;
-  return __brkval ? &top - __brkval : &top - &__bss_end;
-}
+  int free_memory;
+  if (reinterpret_cast<int>(__brkval) == 0) {
+    // if no heap use from end of bss section
+    free_memory = reinterpret_cast<int>(&free_memory)
+                  - reinterpret_cast<int>(&__bss_end);
+  } else {
+    // use from top of stack to heap
+    free_memory = reinterpret_cast<int>(&free_memory)
+                  - reinterpret_cast<int>(__brkval);
+  }
+  return free_memory;
+};
 #endif  // __arm
```

Comment:
__@nophead__ As for leaving things in the project .gitignore file which have nothing to do with the project itself, you are just encouraging "sloppy technique". Everyone who submits a pull request should be correcting the git installation on their machine.

I agree that there will be some confusion while the clueless learn the proper technique.
However, there is no reason that inappropriate files have to be incorporated in the master repository.

We can always "bounce" the proposed commits until someone edits them appropriately.

See http://git-scm.com/docs/gitignore